### DISCUSSION TITLE
Range enlargement

### DISCUSSION THREAD

**Discussion by Vano-ua** [jgromes/RadioLib#507] [Q&A]
Hi. I am using ESP32 + EByte E22 400M30S. I need to transmit 16 bytes every 1-2 sec, but I need range as long as possible. I have set output power to 22 and spreading factor to 12. I know that I can additionnaly enlarge range by setting up air data transmission rate to 0.3 kbit/sec. How can I do it with Radiolib & should I change anby other settings?

**Comment by jgromes** [comment]
E22 400M30S is one of the modules that have external PA - see this wiki page on how to use them: https://github.com/jgromes/RadioLib/wiki/High-power-Radio-Modules-Guide

By setting output from the module to +22 dBm, you are actually saturating the PA input, which will most likely have significant impact on the range you get. Follow the instructions on the page listed above to calculate the correct value to use with external PA

Regarding the bit rate, in LoRa, that is primarily the function of spreading factor and bandwidth. Larger SF + lower BW = more range (usually), however, on extremely narrow BW you have to be careful about frequency calibration and drift of the crystal. It also comes with a significant data rate penalty, it can be as slow as a couple of bits per second.

**Comment by Vano-ua** [comment]
Thanks for Your reply. 
I'm not very good in radio transmission( 
You mean I shouldn't use 22 dBm 'cause 30-12 (power of H0606E PA) = 18, so 18 dBm is maximum and any bigger power will have no effect?
concerning BW - what can You recommend? Or is there any table of BW and corresponding ranges or at least any dependencies?

**Reply by jgromes** [reply]
> You mean I shouldn't use 22 dBm 'cause 30-12 (power of H0606E PA) = 18, so 18 dBm is maximum and any bigger power will have no effect?

Yes, to be more precise, setting higher than 18 dBm output from the SX126x module might have adverse effect, because it might be more than the input of the PA can handle.

> concerning BW - what can You recommend

Without knowing the specifics of your system, I can't recommend anything. It all depends on how much data rate are you willing to sacrifice for longer range. You can find all equations for symbol rates in the SX126x datasheets.

**Comment by Vano-ua** [comment]
I need to send only 15 bytes every 2 seconds, necessary range is 10-20 km. however, I am planning to use external powerful antenna on the receiver - ebyte TX400-BLG-55, it gives 40-80% gain. Transmission will be conducted in non-rural area with minimal interference of other devices.
Besides, what is the minimal effective packet size? I mean what is better for performance - to send 15 bytes every 2 seconds or 30 every 4 seconds?

**Reply by jgromes** [reply]
That antenna is still omni-directional, so I'm a bit sceptical about the 6 dBi gain they claim. If the transmission is always uni-directional, consider adding an external LNA (low-noise amplifier) on the receiver.

With 15 bytes every 2 seconds, you can afford to go quite slow - I would suggest starting with 62.5 kHz bandwidth, which should still be pretty much immune to any temperature-induced changes in the crystal frequency. Make sure you're using the maximum coding rate (4/8) to improve the forward error correction (and by that getting higher link budget margin).

When it comes to effective packet size, you are correct in assuming there is some overhead - preamble, sync word, LoRa header, CRC and the data that gets added by forward error correction. In this case, I think it's going to be larger than your 15 bytes of payload, so if you can afford to only send data once every 4 or 8 seconds (or even slower), you will be using the channel (and power) more efficiently.
